### ISSUE TITLE
Missing documentation on shape functionality requirements

### ISSUE THREAD

**Issue by Mrcubix** [AvaloniaUI/Avalonia#13247]
## Is your feature request related to a problem? Please describe.

A shape has 2 requirements to work properly :

- Overriding `CreateDefiningGeometry`, which is used to create the `Shape`'s `Geometry`,
- The most important requirement : A static constructor calling `AffectGeometry<T>` with its own type with at least `BoundsProperty` as parameter.

```cs
static MyControl()
{
    AffectsGeometry<MyControl>(BoundsProperty);
}
```

If the following requirements aren't met, the shape will not be visible, no matter the styling applied.

## Describe the solution you'd like

One of the following:
- A documentation page on making your own shapes, including the previous requirements,
- Add new line of documentation to AffectGeometry's xmldoc,
- Add new line of documentation to every controls inheriting `Shape`.

## Additional context

Related to #13236, Where shapes weren't appearing due to the previous requirements not being met.

![image](https://github.com/AvaloniaUI/Avalonia/assets/39861216/85485c08-cf70-4453-bd43-cc37ca8dc8ae)

**Comment by timunie** [comment]
This issue should go into docs then. Any contribution is welcome

**Comment by Mrcubix** [comment]
> This issue should go into docs then. Any contribution is welcome

Indeed, just found out the repo, can't transfert the issue on my own it seems like

**Comment by timunie** [comment]
Will do it the other day
